src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNetworkTask.java

```diff
@@ -111,7 +111,8 @@ private void routeItems(Location inputNode, Block inputTarget, int frequency, Ma
                 if (inv.getItem(previousSlot) == null) {
                     inv.setItem(previousSlot, stack);
                 }
-                else {
+                // Making sure the item cannot be added to another slot before dropping it on ground
+                else if((stack = inv.addItem(stack).get(0)) != null){
```

Comment:
This could lead to an IndexOutOfBoundsException. So doing .isEmpty() on the returned map instead would be much safer.

Reply:
You are wrong, that's a Map, not a List. Therefore, running Map#get(0) will return null if everything was fine, not an IndexOutOfBoundsException

Reply:
I even wrote Map but yeah I thought of a Collection, still could be simplified to .isEmpty() nonetheless

Reply:
> 
> 
> I even wrote Map but yeah I thought of a Collection, still could be simplified to .isEmpty() nonetheless

It will just make the code longer, this version works fine and can be easily understood

Reply:
I think the opposite is true.
Doing isEmpty() would be significantly shorter and much easier to read than an assignment and a null check.

Reply:
> 
> 
> I think the opposite is true.
> Doing isEmpty() would be significantly shorter and much easier to read than an assignment and a null check.

I will change it if I have the time...

Reply:
Lemme change it to a "suggested code change" then you only need to hit apply.